### PR TITLE
Browser code editor

### PR DESCRIPTION
**Pull request by jamesnw**
Adds a new page for the Playground, including link and header. 
Adds CodeMirror text editor, with an input and a readonly output, with options for input and output format.

Designs: https://xd.adobe.com/view/84c872fe-0e51-46e4-a59c-f72aab2b8c6a-d19b/screen/f2726fa2-86f8-414d-9e32-66a6a6402918/

### Helpful test links-

1. [Quoted XSS attempts](https://deploy-preview-58--sass-site-oddbird.netlify.app/playground/#MTAlNDBkZWJ1ZyUyMCUyMiUzQ2JvZHklMjBvbmxvYWQlM0RhbGVydCglRTIlODAlOThzb21ldGhpbmclRTIlODAlOTkpJTNFJTIyJTNCJTBBJTQwd2FybiUyMCUyMiUzQ2IlMjBvbm1vdXNlb3ZlciUzRGFsZXJ0KCVFMiU4MCU5OFhTUyUyMHRlc3RpbmchJUUyJTgwJTk4KSUzRSUzQyUyRmIlM0UlMjIlM0IlMEElNDBkZWJ1ZyUyMCUyMiUzQ3NjcmlwdCUzRWFsZXJ0KCdYU1MlMjB0ZXN0aW5nJyklM0MlMkZzY3JpcHQlM0UlMjIlM0IlMEElNDBkZWJ1ZyUyMCUyMiUyNTNjc2NyaXB0JTI1M2VhbGVydCgnWFNTJTIwdGVzdGluZycpJTI1M2MlMkZzY3JpcHQlMjUzZSUyMiUzQiUwQSU0MGRlYnVnJTIwJTIyJTNDYnV0dG9uJTIwb25tb3VzZW92ZXIlM0QlMjJhbGVydCgnaGVjaycpJTIyJTNFVGVzdCUzQyUyRmJ1dHRvbiUzRSUyMiUzQg==)
1. [Unquoted XSS attempts- comment out to run next](https://deploy-preview-58--sass-site-oddbird.netlify.app/playground/#MTAlNDBkZWJ1ZyUyMCUzQ2JvZHklMjBvbmxvYWQlM0RhbGVydCglRTIlODAlOThzb21ldGhpbmclRTIlODAlOTkpJTNFJTNCJTBBJTQwd2FybiUyMCUzQ2IlMjBvbm1vdXNlb3ZlciUzRGFsZXJ0KCVFMiU4MCU5OFhTUyUyMHRlc3RpbmchJUUyJTgwJTk4KSUzRSUzQyUyRmIlM0UlM0IlMEElNDBkZWJ1ZyUyMCUzQ3NjcmlwdCUzRWFsZXJ0KCdYU1MlMjB0ZXN0aW5nJyklM0MlMkZzY3JpcHQlM0UlM0IlMEElNDBkZWJ1ZyUyMCUyNTNjc2NyaXB0JTI1M2VhbGVydCgnWFNTJTIwdGVzdGluZycpJTI1M2MlMkZzY3JpcHQlMjUzZSUzQg==)
1. [Debug mixin, shows same line number repeated](https://deploy-preview-58--sass-site-oddbird.netlify.app/playground/#MTAlNDBtaXhpbiUyMGluc2V0LWRpdmlkZXItb2Zmc2V0KCUyNG9mZnNldCUyQyUyMCUyNHBhZGRpbmcpJTIwJTdCJTBBJTIwJTIwJTI0ZGl2aWRlci1vZmZzZXQlM0ElMjAoMiUyMColMjAlMjRwYWRkaW5nKSUyMCUyQiUyMCUyNG9mZnNldCUzQiUwQSUyMCUyMCU0MGRlYnVnJTIwJTIyZGl2aWRlciUyMG9mZnNldCUzQSUyMCUyMyU3QiUyNGRpdmlkZXItb2Zmc2V0JTdEJTIyJTNCJTBBJTBBJTIwJTIwbWFyZ2luLWxlZnQlM0ElMjAlMjRkaXZpZGVyLW9mZnNldCUzQiUwQSUyMCUyMHdpZHRoJTNBJTIwY2FsYygxMDAlMjUlMjAtJTIwJTIzJTdCJTI0ZGl2aWRlci1vZmZzZXQlN0QpJTNCJTBBJTdEJTBBcCU3QiUwQSUyMCUyMCU0MGluY2x1ZGUlMjBpbnNldC1kaXZpZGVyLW9mZnNldCgxJTJDMiklM0IlMEElMjAlMjAlNDBpbmNsdWRlJTIwaW5zZXQtZGl2aWRlci1vZmZzZXQoMiUyQzIpJTNCJTBBJTdEJTBB)
1. [Indented format](https://deploy-preview-58--sass-site-oddbird.netlify.app/playground/#MDElNDBtaXhpbiUyMGNvcm5lci1pY29uKCUyNG5hbWUlMkMlMjAlMjR0b3Atb3ItYm90dG9tJTJDJTIwJTI0bGVmdC1vci1yaWdodCklMEElMjAlMjAuaWNvbi0lMjMlN0IlMjRuYW1lJTdEJTBBJTIwJTIwJTIwJTIwYmFja2dyb3VuZC1pbWFnZSUzQSUyMHVybCglMjIlMkZpY29ucyUyRiUyMyU3QiUyNG5hbWUlN0Quc3ZnJTIyKSUwQSUyMCUyMCUyMCUyMHBvc2l0aW9uJTNBJTIwYWJzb2x1dGUlMEElMjAlMjAlMjAlMjAlMjMlN0IlMjR0b3Atb3ItYm90dG9tJTdEJTNBJTIwMCUwQSUyMCUyMCUyMCUyMCUyMyU3QiUyNGxlZnQtb3ItcmlnaHQlN0QlM0ElMjAwJTBBJTBBJTBBJTBBJTQwaW5jbHVkZSUyMGNvcm5lci1pY29uKCUyMm1haWwlMjIlMkMlMjB0b3AlMkMlMjBsZWZ0KQ==)
1. [Variety of characters](https://deploy-preview-58--sass-site-oddbird.netlify.app/playground/#MDElNDBkZWJ1ZyUyMCVGMCU5RiU5OCU4MCUwQSU0MGRlYnVnJTIwJUYwJTlGJTkxJUE4JUUyJTgwJThEJUYwJTlGJTkxJUE5JUUyJTgwJThEJUYwJTlGJTkxJUE3JUUyJTgwJThEJUYwJTlGJTkxJUE2JTBBJTQwZGVidWclMjAlQzMlQjElMjAlQzUlQkUlMkMlMjAlQzUlQTElMkMlMjAlQzQlQUYlMkMlMjAlQzQlOTclMkMlMjAlQzUlQUIlMkMlMjAlQzUlQjMlMEElNDBkZWJ1ZyUyMCVFNSVCOSVCMyVFNCVCQiVBRSVFNSU5MCU4RCUwQSU0MGRlYnVnJTIwJUU2JTg4JTkxJTBBJTBBJTI0ZW1vamklM0ElMjAlRjAlOUYlQTQlOTMlMEFwJTBBJTIwJTIwY29udGVudCUzQSUyMCUyNGVtb2ppJTBBJTBB)
1. [Escape char rendering as emoji](https://deploy-preview-58--sass-site-oddbird.netlify.app/playground/#MTElNDBjb3VudGVyLXN0eWxlJTIwdGh1bWJzJTIwJTdCJTBBJTIwJTIwc3lzdGVtJTNBJTIwY3ljbGljJTNCJTBBJTIwJTIwc3ltYm9scyUzQSUyMCUyMiU1QzFGNDREJTIyJTNCJTBBJTdEJTBBJTBBJTBBJTQwZGVidWclMjAlMjIlNUMxRjQ0RCUyMg==) - This is correct per https://github.com/sass/dart-sass/issues/1219
1. [Deprecation and error](https://deploy-preview-58--sass-site-oddbird.netlify.app/playground/#MTElNDBkZWJ1ZyUyMCgxMnB4JTJGNHB4KSUzQiUwQSUwQSU0MGRlYnVnJTIwbWV0YS5tb2R1bGVzKCdwYWdlLXNpemVzJyk=)
1. [Built in modules](https://deploy-preview-58--sass-site-oddbird.netlify.app/playground/#MTElNDB1c2UlMjAlMjJzYXNzJTNBY29sb3IlMjIlM0IlMEElNDB1c2UlMjAlMjJzYXNzJTNBc2VsZWN0b3IlMjIlM0IlMEElMEElNDBkZWJ1ZyUyMGNvbG9yLmFkanVzdCglMjM5OTgwOTklMkMlMjAlMjRsaWdodG5lc3MlM0ElMjAtMzAlMjUlMkMlMjAlMjRhbHBoYSUzQSUyMC0wLjQpJTNCJTIwJTJGJTJGJTIwcmdiYSg3MSUyQyUyMDU3JTJDJTIwNzElMkMlMjAwLjYpJTBBJTQwZGVidWclMjBzZWxlY3Rvci5hcHBlbmQoJTIyLmFjY29yZGlvbiUyMiUyQyUyMCUyMl9fY29weSUyQyUyMF9faW1hZ2UlMjIpJTNCJTBBJTQwZGVidWclMjBzdHJpbmcucXVvdGUoJTIySGVsdmV0aWNhJTIyKSUzQg==)

### CODE REVIEW

File: source/_includes/playground_header.liquid
Comment (jgerigmeyer) [inline source/_includes/playground_header.liquid]: We can wait to see if there are other playground-related includes, but if so then it might be worth nesting this in a new `_includes/playground/` dir.

File: source/_includes/playground_header.liquid
Comment (jgerigmeyer) [inline source/_includes/playground_header.liquid]: Maybe `<h1>` in this case?

File: source/assets/sass/regions/_banner.scss
Comment (jgerigmeyer) [inline source/assets/sass/regions/_banner.scss:41]: Looks fine to me in browser, but I'll defer to @stacyk on style changes.

File: source/_layouts/base.liquid
Comment (jgerigmeyer) [inline source/_layouts/base.liquid]: Is there a good reason for the playground header to be inside of `<main>` instead of up here? If not, I'd suggest something like:

```suggestion
    {% typogr -%}
      {%- if is_playground -%}
        {% render 'playground_header', title: title %}
      {%- else -%}
        {% render 'header', alerts: alerts %}
      {%- endif -%}
    {%- endtypogr %}
```

File: source/_layouts/base.liquid
Comment (jgerigmeyer) [inline source/_layouts/base.liquid]: LiquidJS has some annoying limitations, so I've discovered this syntax won't work for two reasons:

1. Parens inside tags are not respected, so values with filters need to be `{% assign %}`ed first and then used.
2. `or` instead of `||` (see https://liquidjs.com/tutorials/operators.html)

So:

```suggestion
      {% assign is_typedoc = page | isTypedoc %}
      {% unless is_typedoc or is_playground %}
```

File: source/_layouts/base.liquid
Comment (jgerigmeyer) [inline source/_layouts/base.liquid]:
```suggestion
```

File: source/_includes/playground_header.liquid
Comment (jgerigmeyer) [reply source/_includes/playground_header.liquid]: There's almost enough duplicated between this file and `header.liquid` that I'm tempted to combine them with conditionals, but I can see an argument either way. This is probably clearer, so 👍

**Review by jgerigmeyer — COMMENTED** [review]
@jamesnw Looks good! I left a few comments for potential cleanup on the conditionals in `base.liquid`, but I think the general approach is 👍 

You may have already seen it, but note that the PR gets an [automatic comment](https://github.com/oddbird/sass-site/pull/58#issuecomment-1570327910) with a link to the compiled static HTML diff, which can be helpful as a sanity check to be sure the changes aren't generating unexpected markup. (In this case it reveals that the `<h1>` is unintentionally being removed from all pages.)

File: source/_includes/playground_header.liquid
Comment (jamesnw) [reply source/_includes/playground_header.liquid]: Good call, it was more straightforward to override the size in this instance than I thought it would be

File: source/_includes/playground_header.liquid
Comment (jamesnw) [reply source/_includes/playground_header.liquid]: I think it's likely there will be more, so I'll move them over.

File: source/_layouts/base.liquid
Comment (jamesnw) [reply source/_layouts/base.liquid]: My thought was that the playground header was most analogous to the title, so kept it close there. But this code is more straightforward, and I don't think there's a significant semantic reason for that position.

**Review by jamesnw — COMMENTED** [review]
Thanks, @jgerigmeyer! Changes made. The diff checker is nifty!

**Comment by jerivas** [comment]
[Browser code editor & compiled output](https://trello.com/c/XnIoc6or/13-browser-code-editor-compiled-output)

File: source/playground.liquid
Comment (jgerigmeyer) [inline source/playground.liquid]: If we don't want docsearch, this could probably inherit directly from `base` (and could then remove `no_container`).

File: source/assets/js/playground/editor-setup.ts
Comment (jgerigmeyer) [inline source/assets/js/playground/editor-setup.ts]: I think we might want to consider adding this.

File: source/assets/js/playground/playground.ts
Comment (jgerigmeyer) [inline source/assets/js/playground/playground.ts]: Update this when we have a new Dart Sass release.

File: source/playground.liquid
Comment (jamesnw) [reply source/playground.liquid]: Done. I also needed to remove the docsearch JS from the site in order to remove the console error. It was looking for an element to add the search box into, and the header for this page removed the search box.

File: source/assets/js/playground/editor-setup.ts
Comment (jamesnw) [reply source/assets/js/playground/editor-setup.ts]: Added, I agree!

**Comment by jgerigmeyer** [comment]
I added a new `playground` branch and updated this PR to base off that. I think we'll want to maintain this feature-set in its own branch until it's ready to go live, so that we can continue making updates to the rest of the site on `main` without including this feature.

File: source/assets/js/playground.ts
Comment (jgerigmeyer) [inline source/assets/js/playground.ts:5]: @jamesnw I replaced `lodash.debounce` with this, since it should result in the same bundle size and Lodash says the per-method packages are deprecated: https://lodash.com/per-method-packages

**Review by jgerigmeyer — COMMENTED** [review]
Functionality LGTM! @stacyk This is ready for style cleanup.

File: source/assets/js/playground/theme.ts
Comment (jamesnw) [inline source/assets/js/playground/theme.ts:1]: @stacyk Did you happen to take a look at this file?

I think we could move much of the playground theme to /vendor/_codemirror.scss if that better fits the paradigm. The wrapping element would be `.cm-editor` (which is applied with the `&` here).

File: source/assets/sass/components/_playground.scss
Comment (jamesnw) [inline source/assets/sass/components/_playground.scss]: @stacyk This id is targeting the same element as the above class, and I think there's some competition between the two. This is causing the exit animation to appear like this-
image
It immediately loses height, moves to the corner, and then fades out.

File: source/assets/sass/components/_playground.scss
Comment (jamesnw) [inline source/assets/sass/components/_playground.scss]: @stacyk I meant these as placeholders- do we have any brand contextual colors?

File: source/assets/sass/components/_buttons.scss
Comment (jamesnw) [inline source/assets/sass/components/_buttons.scss]: @stacyk  Minor, but it appears that since :focus is after :hover, if you hover while a button has focus, no hover effects are applied. However, switching them makes it so that there is no visual change when a user clicks, it remains in the :hover state. Not sure which is preferred.

File: source/assets/sass/components/_playground.scss
Comment (jamesnw) [inline source/assets/sass/components/_playground.scss]: @stacyk This is a generated class from Code Mirror. Depending on your thoughts regarding where the playground/theme.ts contents should go, we could either define that in the [`&` section there](https://github.com/oddbird/sass-site/pull/58/files#diff-6a330087a8c665d5f7b801855cfddbde64edfd391c57ba61a3e667f5e4c311f8R8) or target `.cm-editor`.

**Review by jamesnw — COMMENTED** [review]
@stacyk This is looking much better, both from the user's view and the code view. Thanks! You didn't break any Javascript connections, but I do have a few minor questions.

File: source/assets/js/playground/theme.ts
Comment (jamesnw) [reply source/assets/js/playground/theme.ts:1]: Per conversation in standup, I'm moving this to playground.scss

File: source/assets/sass/components/_playground.scss
Comment (jamesnw) [inline source/assets/sass/components/_playground.scss]: @stacyk This allows the editor to grow indefinitely with content, instead of scrolling after a certain point. I can go either way on which behavior we want, but if we keep height at 100% than we can remove `.cm-scroller{overflow: auto}` below.

File: source/assets/js/playground/theme.ts
Comment (stacyk) [inline source/assets/js/playground/theme.ts]: remove

File: source/_layouts/base.liquid
Comment (mirisuzanne) [inline source/_layouts/base.liquid:85]: It might make sense to add new layouts that extend base (like a playground-specific layout), rather than putting all of it in the base layout?

File: source/assets/sass/components/_buttons.scss
Comment (stacyk) [inline source/assets/sass/components/_buttons.scss]: add variations to base, then only override with custom properties on these variations of base.

File: source/assets/sass/components/_buttons.scss
Comment (stacyk) [inline source/assets/sass/components/_buttons.scss]: default to inline-block

File: source/assets/sass/components/_buttons.scss
Comment (stacyk) [inline source/assets/sass/components/_buttons.scss]: remove after updating default button display

File: source/assets/sass/components/_playground.scss
Comment (stacyk) [inline source/assets/sass/components/_playground.scss]: will remove when I write the playground header grid styles

File: source/assets/sass/components/_playground.scss
Comment (stacyk) [inline source/assets/sass/components/_playground.scss]: consider having sass take up full height, and css and debug stack on right.

**Comment by jerivas** [comment]
@stacyk the new layout looks great!

@jamesnw (and others) some random thoughts:

- Could we display the Sass version somewhere? Maybe in the top right corner of the main panel, or logged to the console? You should be able to get it as `sass.info` but it will probably require parsing in some way. Here's the [source](https://github.com/sass/dart-sass/blob/cb4b442408ca9921c21f320485f889463b538e16/lib/src/node.dart#L45-L49) if it helps.
- When I get an autocomplete suggestion and hit `Tab`, the code is indented instead of inserting the suggestion. I find this unexpected because it's different from VS Code and the TS playground.

**Comment by jamesnw** [comment]
> * Could we display the Sass version somewhere? Maybe in the top right corner of the main panel, or logged to the console? You should be able to get it as `sass.info` but it will probably require parsing in some way. Here's the [source](https://github.com/sass/dart-sass/blob/cb4b442408ca9921c21f320485f889463b538e16/lib/src/node.dart#L45-L49) if it helps.

@jerivas That is a great idea, but brings up a second question- how will the sass version be kept up to date for the deployed site? As a user, I would expect that the playground is always at the latest version, but that won't be automatic. 

If it _were_ always in sync with the published version, we could just use the existing release global data. But if it is out of sync (even by a day or more likely a week) we would want to make sure that those values are pulled separately.

**Comment by jamesnw** [comment]
> * When I get an autocomplete suggestion and hit `Tab`, the code is indented instead of inserting the suggestion. I find this unexpected because it's different from VS Code and the TS playground.

@jerivas Tab is tricky, as the base behavior for Codemirror is to allow the user to use tab to move focus out of the editor. We are overriding that to allow the user to use tab which is especially important for the indented mode. (The user can still hit `esc+Tab` to move out of the editor). Potentially we could vary the behavior if autocomplete is open, but I haven't looked into feasability.

File: source/_layouts/base.liquid
Comment (jgerigmeyer) [reply source/_layouts/base.liquid:85]: I'm not sure it's worth it in this case -- it would involve various conditionals and content blocks either way. I'd say this is arguably clearer?

**Comment by jgerigmeyer** [comment]
> > * Could we display the Sass version somewhere? Maybe in the top right corner of the main panel, or logged to the console? You should be able to get it as `sass.info` but it will probably require parsing in some way. Here's the [source](https://github.com/sass/dart-sass/blob/cb4b442408ca9921c21f320485f889463b538e16/lib/src/node.dart#L45-L49) if it helps.
> 
> @jerivas That is a great idea, but brings up a second question- how will the sass version be kept up to date for the deployed site? As a user, I would expect that the playground is always at the latest version, but that won't be automatic.
> 
> If it _were_ always in sync with the published version, we could just use the existing release global data. But if it is out of sync (even by a day or more likely a week) we would want to make sure that those values are pulled separately.

Good point. We definitely want to display the used version (which is what you have now), not necessarily the latest version. I'm a bit nervous about any system that automatically updates the playground version to the latest release without user intervention/approval, since that new release could potentially introduce breaking changes. But we could _warn_ on build if the two versions don't match? I added a note to our next mtg agenda card, to discuss this with Natalie.

**Comment by jerivas** [comment]
> how will the sass version be kept up to date for the deployed site?

Since `sass` is part of the dependencies in `package.json`, I guess it would be updated as part of general dependency upgrades, which at the time is handled by dependabot. Of course, the version can also be upgraded manually at any time to match a new Sass release.

> we could vary the behavior if autocomplete is open, but I haven't looked into feasability

That's fine, I don't think it's super important, just something that jumped out to me while trying things out

File: source/assets/js/playground.ts
Comment (jgerigmeyer) [inline source/assets/js/playground.ts]: @jamesnw Might be worth adding a comment here for why the `?` is used?

Comment (jamesnw) [reply source/assets/js/playground.ts]: After investigating, this isn't actually making a difference, so I reverted back to no query.

**Comment by stacyk** [comment]
Styles are ready for review @mirisuzanne at your convenience

File: source/_includes/footer.liquid
Comment (mirisuzanne) [inline source/_includes/footer.liquid]: Seems like this class could be added to the existing `div` around these links? Nut maybe I'm missing something (and it's not terribly important)

File: source/assets/sass/components/_playground.scss
Comment (mirisuzanne) [inline source/assets/sass/components/_playground.scss]: Not specific to this branch, but consider moving `_breakpoints.scss` into the config folder - since it doesn't have css output of it's own? Same with `_functions.scss`?

File: source/assets/sass/components/_buttons.scss
Comment (mirisuzanne) [inline source/assets/sass/components/_buttons.scss]: Consider splitting `visual-design/typography` into a typography config (with the things that get used in various places) and a typography partial that outputs CSS, and only needs to be used in the final output?

File: source/assets/sass/components/_playground.scss
Comment (mirisuzanne) [inline source/assets/sass/components/_playground.scss:140]: When we have named areas like this, I'd have some preference for making sure each grid item is assigned to an area (like this one is, but others aren't) – even when it's not technically necessary for the layout. By assigning each element to an area, we can ensure that changes to the grid-template have the desired impact.

**Review by mirisuzanne — APPROVED** [review]
Looking great!

File: source/assets/sass/components/_playground.scss
Comment (stacyk) [reply source/assets/sass/components/_playground.scss:140]: They are all using a named grid area, but they are lower in the file... 

`editor-tabbar` is line 285 currently
`editor-error` is 241

Is that what you mean @mirisuzanne?

File: source/assets/sass/components/_buttons.scss
Comment (stacyk) [reply source/assets/sass/components/_buttons.scss]: Added this suggestion to the card: https://trello.com/c/V2ujZ5Pc/5-split-config-only-css-from-the-css-that-has-output

File: source/assets/sass/components/_playground.scss
Comment (stacyk) [reply source/assets/sass/components/_playground.scss]: https://trello.com/c/V2ujZ5Pc/5-split-config-only-css-from-the-css-that-has-output Added this suggestion to the card checklist ... but I probably can quickly take care of this.

File: source/_includes/footer.liquid
Comment (stacyk) [reply source/_includes/footer.liquid]: this is done in https://github.com/oddbird/sass-site/pull/58/commits/e270b5c0dc37e277aab3ab0cd2fafa5629ceb8c9

File: source/assets/sass/components/_playground.scss
Comment (stacyk) [reply source/assets/sass/components/_playground.scss]: done: https://github.com/oddbird/sass-site/pull/58/commits/31d9b56e1e9cc6e0ff132c926354dc7f6f82bb2e

File: source/assets/sass/components/_buttons.scss
Comment (stacyk) [reply source/assets/sass/components/_buttons.scss]: done in https://github.com/oddbird/sass-site/pull/58/commits/31d9b56e1e9cc6e0ff132c926354dc7f6f82bb2e

**Comment by stacyk** [comment]
[Split Config only CSS from the CSS that has output](https://trello.com/c/V2ujZ5Pc/5-split-config-only-css-from-the-css-that-has-output)

File: source/assets/sass/components/_playground.scss
Comment (mirisuzanne) [reply source/assets/sass/components/_playground.scss:140]: oh, yes, ok - I missed that. 👍🏼

**Review by jgerigmeyer — APPROVED** [review]
Looks good to me! @mirisuzanne Back to you for final review

**Review by mirisuzanne — APPROVED** [review]
Looks great!

probably a different story, but would be useful to open/close the console - so it's out of the way when it's not needed.

**Comment by stacyk** [comment]
> Looks great!
> 
> probably a different story, but would be useful to open/close the console - so it's out of the way when it's not needed.

Agreed @mirisuzanne  https://trello.com/c/CmSUBZBC/53-stretch-or-expand-collapse-input-output-panel-widths I believe that would be part of the design/proposal for this collapse/expand panels story.

**Review by stacyk — APPROVED** [review]

**Pull request merged**
